Update tutorial

victorlin opened the pull request:
[_preview_](https://github.com/nextstrain/veme-2022/tree/victorlin/2025-refresh)

## Description of proposed changes

Many things have changed since this tutorial was last used. About half of the the commits are trivial updates, and the other half are content-related changes. Let me know if you'd like to review these separately – I can easily split into separate PRs.

## Checklist

- [x] Finish draft commits
- [ ] Continue updating terminology
- [x] Rename repo to "nextstrain-walkthrough"?

victorlin committed:
Use nextstrain-base Conda package for Binder

This package provides continually updated versions of Nextstrain
software and dependencies, aligning with GitHub Codespaces' use of the
nextstrain/base Docker image.

victorlin committed:
Update URLs for repo transfer

victorlin committed:
Replace Nextalign CLI with Nextclade CLI

Following the doc page on Nextclade v3 migration.¹

¹ <https://docs.nextstrain.org/projects/nextclade/en/stable/user/migration-v3.html>

victorlin committed:
Set up for GitHub Codespaces

This is a remote-hosted option that works well for this tutorial.

<https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/setting-up-your-repository/facilitating-quick-creation-and-resumption-of-codespaces#creating-an-open-in-github-codespaces-badge>

victorlin committed:
Adjust wording from "part of the workflow" to "step"

The tutorial is explicitly not creating a workflow, rather it is listing
out a series of steps. Adjust wording accordingly.

victorlin committed:
Name the Auspice dataset "nextstrain-walkthrough"

Make this more distinct from the existing "example-ncov" dataset, and
more relevant to the overall context of this guide. I briefly considered
"nextstrain-tutorial", but that seemed too generic given our other
existing tutorials.

victorlin committed:
Remove GitPod option

GitPod-hosted environments are now only available for users who have
joined before October 1, 2024 on non-Enterprise plans, and will no
longer be an option in fall 2025. The GitPod Flex option requires a
self-hosted cloud account which is impractical for this tutorial.

<https://www.gitpod.io/flex-or-classic>

victorlin committed:
Replace "Auspice JSON" with "Auspice dataset"

It's more common for Nextstrain docs and pages to use the term
"dataset", so users would benefit from learning the term in this
tutorial.

victorlin committed:
Use Auspice interface throughout the tutorial

This allows for incremental familiarity with the Auspice interface,
going from a very limited view of tree_raw.nwk to a wide range of panels
and options available by the end of the tutorial. It is also nice to see
visual differences between different steps.

Screenshots generated from commands run with a random seed of 0 (`augur
filter --subsample-seed 0`, `augur tree --tree-builder-args="-seed 0"`,
`augur refine/ancestral --seed 0`).

victorlin committed:
Update license file

Drop the year and put things under the name of Nextstrain.

victorlin committed:
Update example data

Make the data representative of recent evolution.

victorlin committed:
Use HTTPS links to auspice.us, nextstrain.org

HTTP → HTTPS redirects are in place, but it's better to have the direct
link.

victorlin committed:
Put Wuhan-Hu-1/2019 at top of FASTA files

Reasoning in README change. The root of IQ-TREE's tree_raw.nwk does not
matter for this tutorial as-is, but I am planning to update the tutorial
to explicitly inspect tree_raw.nwk with auspice.us, and it would be nice
to have an accurate root.

victorlin commented:
@huddlej @j23414 tagging you for review since you've previously used this to run the show down at VEME. It's not ready for merge, but I'm in the process of changing things quite a bit with 96f464c536b11b77c838d2d3820b317fa4081e16 and thought it would be worth gathering any feedback about the direction now before making more changes. Happy to chat about this in person too.

genehack commented on README.md:
```diff
@@ -2,15 +2,14 @@
 
   - [Install Nextstrain](https://docs.nextstrain.org/en/latest/install.html) on your computer or launch one of the virtual machine providers listed above.
-  - Clone this repository with `git clone https://github.com/blab/veme-2022.git` and change into the resulting `veme-2022` directory.
+  - Clone this repository with `git clone https://github.com/nextstrain/veme-2022.git` and change into the resulting `veme-2022` directory.
```
This is probably a bigger discussion than is appropriate for the bounds of this PR, but: perhaps we should consider renaming the repo to something more descriptive, more generic, and less dated looking?

victorlin replied:
The scope of this PR is intentionally broad as "update everything", so I think good to discuss here. My proposed rename is "nextstrain-walkthrough" with the same reasoning as in 6fd86ec1af527fa3dec8cbaab24e0bec335a561b.

genehack replied:
+1 for `nextstrain-walkthrough` ...but i will see your bikeshed and raise you `nextstrain-tour`?

victorlin replied:
Hmm... when I think "tour", I imagine something more hands-off, like visiting all the existing analyses on nextstrain.org and pointing out the datasets/curated data that we have made available. "walkthrough" seems more fitting for the hands-on step-by-step tutorial here.

Thinking about the various outreach projects I'm preparing, "Nextstrain tour" is fitting for both the 30 minute webinar and conference poster, while "Nextstrain walkthrough" is fitting for the 90 minute workshop.

genehack replied:
> Hmm... when I think "tour", I imagine something more hands-off, like visiting all the existing analyses on nextstrain.org and pointing out the datasets/curated data that we have made available. "walkthrough" seems more fitting for the hands-on step-by-step tutorial here.

Fair enough — at least part of my motivation was just trying to come up with something shorter and maybe less prone to misspelling than "through" — I always have to think about that, just for a second.

victorlin replied:
"walkthru"? 😆 maybe after using it 100 times it'll feel normal like "drive thru"...

victorlin replied:
from https://github.com/nextstrain/veme-2022/pull/6#pullrequestreview-2751470517:

> “nextstrain-walkthrough” or “tour” etc. don’t tell me much about the contents or scope of the tutorial. What if we consider the name of these materials in the context of the official docs where these will be a tutorial? In that context, how would you describe the materials?

Great point. I think the scope is well defined in the README already:

https://github.com/nextstrain/veme-2022/blob/d187ba0da5246af45e8e9cce1f2b2dad38259366/README.md#L16-L26

I previously mentioned in https://github.com/nextstrain/docs.nextstrain.org/pull/244#discussion_r2017638014 that this tutorial is a rehashed version of "[Creating a phylogenetic workflow](https://docs.nextstrain.org/en/latest/tutorials/creating-a-phylogenetic-workflow.html)", but that's only partially true. While there is lots of overlap (filter/align/refine/traits/ancestral/export), this tutorial has additional frequencies and clades (maybe simply due to example data being SARS-CoV-2?) while "Creating a phylogenetic workflow" brings Snakemake into the mix, though not in detail.

SARS-CoV-2 is used for example data but I don't think it should be mentioned in the tutorial name – we have existing SARS-CoV-2 tutorials which are about using the ncov workflow, a much different scope. However, not mentioning the pathogen makes it harder to differentiate this one and "Creating a phylogenetic workflow". When putting this new tutorial into the official docs, we may have to rethink/rename some of the other existing tutorials. I'll think more about this...

huddlej replied:
I've thought for a while now that we could replace the current "creating a phylogenetic workflow" tutorial with this VEME content and talk about the more advanced aspects of building a Snakemake workflow in a different tutorial.

victorlin replied:
I was thinking that too, but as more than just a direct replacement:

- I'd like to reconsider the name "creating a phylogenetic workflow", at least by dropping the term "workflow" since that's beyond the scope.
- "Running a phylogenetic workflow" shouldn't point to this with the existing phrasing of "learn how to create the workflow in this tutorial", unless we want to replace the underlying data and workflow used in that tutorial.
- Maybe this is too much to do now, but I'd also like to replace the current "running a phylogenetic workflow" tutorial. I'm not sure what purpose it serves now other than familiarizing users with `git clone`+`nextstrain build`+`nextstrain view`, since none of our existing workflows can be run simply by these 3 commands. At the very least, it should change soon with the release of `nextstrain run`.

victorlin replied:
I've renamed as `nextstrain-walkthrough` for now. While it's not perfect, I think it's at least more self-descriptive than `veme-2022` and will be useful for initial sharing of the draft.

victorlin replied:
Continuing conversation in https://github.com/nextstrain/nextstrain-walkthrough/issues/9

genehack commented on README.md:
```diff
@@ -204,16 +203,24 @@ Infer a divergence tree from the alignment.
+
+![tree_raw.nwk in auspice.us](images/tree_raw.jpeg)
+
+Let's familiarize with the Auspice interface. The main view is an interactive phylogenetic tree. To the left are controls for the main view. There is not much going on because Newick files lack additional data that powers most of the Auspice interface. We will add those in the next steps.
```
```suggestion
Let's familiarize ourselves with the Auspice interface. The main view is an interactive phylogenetic tree. To the left are controls for the main view. There is not much going on with this particular visualization, because Newick files lack the additional data that powers most of the Auspice interface. We will add that in the next steps.
```

victorlin replied:
Added in d187ba0da5246af45e8e9cce1f2b2dad38259366

genehack commented on README.md:
````diff
@@ -363,24 +387,16 @@ augur export v2 \
 ```
 
-Then, you can view the trees by navigating to http://localhost:4000 or the corresponding address from GitPod.
+View the tree with additional metadata in Auspice. It should look something like below. Note the newly added info.
````
there's no "below"?

victorlin replied:
Oops. I had written the text part of 96f464c536b11b77c838d2d3820b317fa4081e16 and started adding screenshots, then decided I should post this for review first before finishing that commit. I'll still wait for more commentary, but the general idea is that there will be screenshots after each `augur export` command. I've added placeholders in d187ba0da5246af45e8e9cce1f2b2dad38259366 so that work in progress is more obvious.

genehack reviewed (commented):
all comments are non-blocking suggestions.

j23414 commented on README.md:
```diff
@@ -2,10 +2,9 @@
 ## Prerequisites
 
-If you want to use a preconfigured environment, you only need to launch one of the following virtual machine providers.
+Use the button below to launch a preconfigured environment in your web browser or Visual Studio Code.
```
[non-blocking, just thinking about supporting multiple compute vs only having one in the tutorial]

Looking online, I agree that GitHub Codespaces seems more robust than Binder. Just a note that Codespaces may require workshop attendees to have a GitHub account while Binder does not. This might be fine if we want to require all workshop attendees to have a GitHub account. 

The other thing to consider is cost (hopefully free). Codespaces seems to have a [cap on compute before it requires switching to a pro account](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#monthly-included-storage-and-core-hours-for-personal-accounts). Binder has a [per session memory cap but otherwise seems free](https://mybinder.readthedocs.io/en/latest/about/user-guidelines.html#how-much-memory-am-i-given-when-using-binder).  

My tendency is to provide both (to give people options or certain options change in the future) but recommend one (GitHub Codespaces) during a workshop. If they've reached the cap of Codespaces, they can try Binder. However, feel free to tailor this for your workshop audience.

victorlin replied:
Fair point! I'll put Binder back (revert 0a2fd2faad799777a35b36d185aa570894007341) but will ask my workshop attendees to use GitHub Codespaces if they are able to.

victorlin replied:
from https://github.com/nextstrain/veme-2022/pull/6#pullrequestreview-2751470517:

> I second Jennifer’s concerns about having alternate platforms for learners to run the workshop commands. I [wrote some pros/cons of different platforms back in 2022](https://docs.google.com/document/d/1OPQ9rffXt0JlZT25FJP9QRvZYF8YSdEOtLMFxXdGrvU/edit?tab=t.0#heading=h.s4uu9o96hv9e) which might be worth updating and documenting here in some way for future us. Binder timeouts were an issue for me back then, too.

Thanks, I forgot about that Google doc! I can see some sections of that doc living in an "internal" README in this repo. Do you mind if I make edits/comments on the doc, then move things over to this repo later?

victorlin replied:
I've replaced 0a2fd2faad799777a35b36d185aa570894007341 with d14761b which explains the 3 options (GitHub codespace, Binder, local installation) in more detail.

huddlej replied:
> Do you mind if I make edits/comments on the doc, then move things over to this repo later?

For sure. Thanks!

huddlej reviewed (commented):
* Was the [update to example data](https://github.com/nextstrain/veme-2022/pull/6/commits/21a2e41a9f52fe1dec74a2e37c716bb965c4f08b) mostly to make the examples more representative of recent evolution or were there issues with the original example data?
* I’d avoid using awk to manipulate FASTA files in favor of tools like seqkit that are designed for working with FASTA files. For example, you could replace the awk commands with two seqkit commands like:

```
seqkit grep -n -p "Wuhan-Hu-1/2019" sequences_unordered.fasta > sequences.fasta
seqkit grep -v -n -p "Wuhan-Hu-1/2019" sequences_unordered.fasta >> sequences.fasta
```

* I second Jennifer’s concerns about having alternate platforms for learners to run the workshop commands. I [wrote some pros/cons of different platforms back in 2022](https://docs.google.com/document/d/1OPQ9rffXt0JlZT25FJP9QRvZYF8YSdEOtLMFxXdGrvU/edit?tab=t.0#heading=h.s4uu9o96hv9e) which might be worth updating and documenting here in some way for future us. Binder timeouts were an issue for me back then, too.
* I love the use of the nextstrain-base Conda package for Binder. Our tools have come so far since 2022!
* “nextstrain-walkthrough” or “tour” etc. don’t tell me much about the contents or scope of the tutorial. What if we consider the name of these materials in the context of the official docs where these will be a tutorial? In that context, how would you describe the materials?
* I like the idea of introducing Auspice incrementally throughout the workshop. Once you’ve finished your first pass at revising the contents, you may want to practice the whole workshop like you would give it in person and time yourself. You’ll find bits of the tutorial you need to edit for time or clarity that the rest of us won’t detect just by silently reading the contents to ourselves.

victorlin commented on data/README.md:
Re: https://github.com/nextstrain/veme-2022/pull/6#pullrequestreview-2751470517

> Was the [update to example data](https://github.com/nextstrain/veme-2022/pull/6/commits/21a2e41a9f52fe1dec74a2e37c716bb965c4f08b) mostly to make the examples more representative of recent evolution or were there issues with the original example data?

Sorry, I forgot to explain why in the commit message. Purely to make examples more representative of recent evolution.

victorlin replied:
Updated commit message: 42c4112

victorlin commented on data/README.md:
```diff
@@ -9,10 +9,24 @@ nextclade dataset get --name sars-cov-2 --output-dir data
+         /^>Wuhan-Hu-1\/2019/ {p=1; next}
+         /^>/ && p==1 {p=0}
+         p==0 {print}' data/sequences-unordered.fasta) \
+> data/sequences.fasta
```
Re: https://github.com/nextstrain/veme-2022/pull/6#pullrequestreview-2751470517

> I’d avoid using awk to manipulate FASTA files in favor of tools like seqkit that are designed for working with FASTA files. For example, you could replace the awk commands with two seqkit commands like:
> 
> ```
> seqkit grep -n -p "Wuhan-Hu-1/2019" sequences_unordered.fasta > sequences.fasta
> seqkit grep -v -n -p "Wuhan-Hu-1/2019" sequences_unordered.fasta >> sequences.fasta
> ```

Thanks, I will go with those `seqkit` commands. I got the `awk` commands from asking ChatGPT to "generate a bash command" for this. Looking back, I should've explicitly allowed using third party tools in the prompt (I just re-prompted with that clarification and it gave the exact same `seqkit` commands).

victorlin replied:
Updated:

https://github.com/nextstrain/veme-2022/blob/d66ebfc9fa4fc9e5e973612989ba3b1be10ec2d2/data/README.md#L21-L22

victorlin commented on README.md:
Re: https://github.com/nextstrain/veme-2022/pull/6#pullrequestreview-2751470517

> I second Jennifer’s concerns about having alternate platforms for learners to run the workshop commands. I [wrote some pros/cons of different platforms back in 2022](https://docs.google.com/document/d/1OPQ9rffXt0JlZT25FJP9QRvZYF8YSdEOtLMFxXdGrvU/edit?tab=t.0#heading=h.s4uu9o96hv9e) which might be worth updating and documenting here in some way for future us. Binder timeouts were an issue for me back then, too.

Thanks, I forgot about that Google doc! I can see some sections of that doc living in an "internal" README in this repo. Do you mind if I make edits/comments on the doc, then move things over to this repo later?

victorlin replied:
Copied to the relevant thread: https://github.com/nextstrain/veme-2022/pull/6#discussion_r2036137839

victorlin reviewed (commented):
Thanks for the review @huddlej! Added follow-up comments in separate threads.

victorlin committed:
Explain environment options in prerequisites section

Help users choose between the different options and state the
limitations upfront.

victorlin committed:
Update clades.tsv

This is ncov's latest clades.tsv¹ mixed with WHO labels up to 21L
(Omicron) for added context.

¹ <https://github.com/nextstrain/ncov/blob/14fbd7287d270705df9d9fb68adcdaa1d3c926e5/defaults/clades.tsv>

victorlin commented:
I'm going to merge this as a "good enough" first pass update. This is so I can share the initial draft as just the repo URL, then I'll create another PR for further updates.

I'm also going to rename to `nextstrain-walkthrough` for now. While it's not perfect, I think it's at least more self-descriptive than `veme-2022`.
